packages/ragbits-core/src/ragbits/core/vector_stores/_cli.py

```diff
@@ -0,0 +1,117 @@
+import asyncio
+from dataclasses import dataclass
+from pathlib import Path
+
+import typer
+from pydantic import BaseModel
+from rich.console import Console
+
+from ragbits.cli import cli_state, print_output
+from ragbits.cli.state import OutputType
+from ragbits.core.config import core_config
+from ragbits.core.embeddings.base import Embeddings
+from ragbits.core.utils.config_handling import InvalidConfigError
+from ragbits.core.vector_stores.base import VectorStore, VectorStoreOptions
+
+vector_stores_app = typer.Typer(no_args_is_help=True)
+
+
+@dataclass
+class CLIState:
+    vector_store: VectorStore | None = None
+
+
+state: CLIState = CLIState()
+
+
+@vector_stores_app.callback()
+def common_args(
+    factory_path: str | None = None,
+    yaml_path: str | None = None,
+) -> None:
+    try:
+        state.vector_store = VectorStore.subclass_from_defaults(
+            core_config,
+            factory_path_override=factory_path,
+            yaml_path_override=Path.cwd() / yaml_path if yaml_path else None,
+        )
+    except InvalidConfigError as e:
+        Console(stderr=True).print(e)
+        raise typer.Exit(1) from e
+
+
+@vector_stores_app.command(name="list")
+def list_entries(limit: int = 10, offset: int = 0) -> None:
```

Comment:
list entries by default should not display vectors - it's not readable that way. Let's add a flag for that (--show-vectors? by default `false`)

Reply:
We agreed to do this in separate PRs (but the same ticket)